ceph/rados/pool_workflows.py

```diff
@@ -683,3 +683,221 @@ def fetch_pool_stats(self, pool: str) -> dict:
+        test_pg_merge = kwargs.get("test_pg_merge", True)
+        modify_threshold = kwargs.get("modify_threshold", False)
+
+        if overwrite_recovery_threads:
+            self.rados_obj.change_recovery_threads(config={}, action="set")
+
+        autoscale_status = self.rados_obj.get_pg_autoscale_status(pool_name=pool)
+        if autoscale_status["pg_autoscale_mode"] != "on":
+            log.debug(
+                f"PG autoscale mode of the pool is not on. Status of pool : {autoscale_status}"
+                f"Turning the autoscale mode to ON for pool : {pool}"
+            )
+            self.rados_obj.set_pool_property(
+                pool=pool, props="pg_autoscale_mode", value="on"
+            )
+
+        init_pg_count = self.rados_obj.get_pool_property(pool=pool, props="pg_num")[
+            "pg_num"
+        ]
+        log.debug(
+            f"init PG count on the pool {pool} before starting any activity is: {init_pg_count}"
+        )
+
+        # Checking if PG increase is desired
+        if test_pg_split:
+            # checking if bulk is already set on pool
+            if self.rados_obj.get_pool_property(pool=pool, props="bulk")["bulk"]:
+                log.error(f"Bulk flag already set on pool : {pool}")
+                return False, 0
+
+            log.info(
+                f"Starting to split the PGs of the pool {pool} by setting bulk flag"
+            )
+            self.rados_obj.set_pool_property(pool=pool, props="bulk", value="true")
+            time.sleep(10)
+
+            # checking if bulk was set successfully
+            if not self.rados_obj.get_pool_property(pool=pool, props="bulk")["bulk"]:
+                log.error(f"Bulk flag could not be set on pool : {pool}")
+                return False, 0
+
+            pg_count_bulk_true = self.rados_obj.get_pool_details(pool=pool)[
+                "pg_num_target"
+            ]
+            log.debug(
+                f"PG count on pool {pool} post addition of bulk flag : {pg_count_bulk_true}"
+            )
+            no_count_change = False
+            if pg_count_bulk_true <= init_pg_count:
+                log.debug(
+                    "PG count before and after bulk flag is same. Checking the threshold values for pool"
+                )
+                log.info(
+                    "Checking the ideal PG count on the pool and the threshold set. default is 3"
+                )
+                pool_data = self.rados_obj.get_pg_autoscale_status(pool_name=pool)
+                final_pg_num = pool_data["pg_num_final"]
+                log.debug(
+                    f"Pool : {pool} details as fetched from autoscale-status : {pool_data}"
+                )
+                # for the threshold var, refer :
+                # https://docs.ceph.com/en/latest/rados/operations/placement-groups/#viewing-pg-scaling-recommendations
+                if float(final_pg_num / init_pg_count) <= 3.0:
```

Comment:
Could you please explain the logic about the threshold value?  
I checked another document to understand the details.
https://ceph.io/en/news/blog/2022/autoscaler_tuning/

Reply:
What are the queries that you have with regard to Threshold then? The explanation given in both the doc atatched as comment and the doc that you have attached provide sufficient explanation as to what does the threshold value does.